crates/polars-lazy/src/physical_plan/executors/scan/csv.rs

```diff
@@ -12,16 +12,13 @@ pub struct CsvExec {
 
 impl CsvExec {
     fn read(&mut self) -> PolarsResult<DataFrame> {
-        let mut with_columns = mem::take(&mut self.file_options.with_columns);
-        let mut projected_len = 0;
-        with_columns.as_ref().map(|columns| {
-            projected_len = columns.len();
-            columns
-        });
+        let with_columns = self
```

Comment:
Can we leave a comment here? It is still not exactly clear to me what we are trying to do. Looking at previous code I guess we try set `with_columns` to `None` if we don't project any columns.

Reply:
That is what it looks like to me too. Sure I'll add a comment.

Reply:
What does `with_columns = None` mean? Does that mean "no specific columns", i.e. all columns, or no columns? Is it different from `Some(v)` where `v` is empty?

Do we have a reason to distinguish between the two? I can imagine that a "row count" operation does not need to store data from any columns. However, if the csv reader then turns `Some(Vec::new())` into `None`, do we end up reading all columns?

Reply:
It means we don't do an explicit column projection, thus yes, select all columns.

`Some(vec![])` would be no columns.

Reply:
So it's impossible to explicitly select no columns with the csv reader, and that is how it should be?

Reply:
> So it's impossible to explicitly select no columns with the csv reader, and that is how it should be?

What would the use-case of that be? I think we never have hit that case because it has zero utility.

Reply:
@ritchie46 added a comment saying 

```
            // Interpret selecting no columns as selecting all columns.
```